### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by jhpratt**
Found using the [`typos-cli` crate](https://crates.io/crates/typos-cli). There are no material changes in this PR.

### CODE REVIEW

File: text/0550-macro-future-proofing.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ possible to lift these restrictions backwards compatibly if desired.
 
 # Key Terminology
 
-- `macro`: anything invokable as `foo!(...)` in source code.
+- `macro`: anything invocable as `foo!(...)` in source code.
```
Comment (lebensterben) [inline text/0550-macro-future-proofing.md:15]: invocable is an alternate spelling of invokable so this is not a typo.

File: text/3231-cargo-asymmetric-tokens.md [hunk 2]
```diff
@@ -301,7 +301,7 @@ It will then derive:
...
+v3.public.eyJjaGFsbGVuZ2UiOiAiY2hhbGxlbmdlIiwgIm11dGF0aW9uIjogInB1Ymxpc2giLCAibmFtZSI6ICJmb28iLCAidmVycyI6ICIwLjAuMCIsICJja3N1bSI6ICJmN2RiYjZhY2ZlZmYxZDQ5MGZiYTY5M2E0MDI0NTZmNzZiMzQ0ZmVhNzdhNWU3Y2FlANDNiNTk3MGMzMzMyYjhmIiwgInN1YiI6ICJwcml2YXRlLWtleS1zdWJqZWN0IiwgImlhdCI6ICIyMDIyLTAyLTI4VDE4OjMzOjI0KzAwOjAwIn36ifmVYCSBYcjHVjQ_JD6R16dcWPEjHYVFOR7QRx3riOLiH7o-m236uNs2NEu-NzOCDZZbsVXvxhop-aUKRc9D-jphV5KFuC8y6mNLklfg1PpH37QeDsyzJDZy604gZ5c.eyJ1cmwiOiAiaHR0cHM6Ly9yZWdpc3RyeS1jaGFsbGVuZ2Utc3ViamVjdC5jb20vY3JhdGUtaW5kZXgiLCAia2lkIjogImszLnBpZC5RQjNXTkJQLTVqLTBYUVYyTU91dnVPY0xsSjh1ei1wbXF0SVp1czF4M1lUdSJ9
```
Comment (lebensterben) [inline text/3231-cargo-asymmetric-tokens.md:304]: not a typo

File: text/2945-c-unwind-abi.md [hunk 3]
```diff
@@ -261,7 +261,7 @@ In order to limit the scope of this RFC, the following limitations are imposed:
 * As noted in the [summary][summary], if a Rust frame containing a pending
   `catch_unwind` call is unwound by a foreign exception, the behavior is
   undefined for now.
-* The behavior of asynchronous exceptions, such as SEH on Windows, interrupting
```
Comment (lebensterben) [inline text/2945-c-unwind-abi.md:264]: SEH is Structured Exception Handling

not a typo

File: text/2696-debug-map-key-value.md [hunk 4]
```diff
@@ -248,7 +248,7 @@ Another alternative is to avoid panicking if the sequence of entries doesn't fol
 # Prior art
 [prior-art]: #prior-art
 
-The `serde::ser::SerializeMap` API (and `libserialize::Encoder` for what it's worth) requires map keys and values can be serialized independently. `SerializeMap` provides a `serialize_entry` method, which is similar to the existing `DebugMap::entry`, but is only supposed to be used as an optimization.
```
Comment (lebensterben) [inline text/2696-debug-map-key-value.md:251]: not a typo

File: text/1620-regex-1.0.md [hunk 5]
```diff
@@ -633,7 +633,7 @@ regexes by default. Go is the only language this author knows of whose standard
 regex implementation guarantees linear time matching. Of course, RE2
 is also worth mentioning, which is a C++ regex library that guarantees linear
 time matching. There are other implementations of regexes that guarantee linear
-time matching (TRE, for example), but none of them are particularly popular.
+time matching (TREE, for example), but none of them are particularly popular.
```
Comment (lebensterben) [inline text/1620-regex-1.0.md:636]: not a typo

File: text/0550-macro-future-proofing.md (hunk 1 above)
Comment (jhpratt) [reply text/0550-macro-future-proofing.md:15]: Merriam-Webster does not list "invokable" as a word. While some sources do list it, Google ngrams shows "invocable" has been consistently used roughly twice as much over the past couple decades (6x for American English, which Rust supposedly prefers). Given these two facts, I think "invocable" is the choice we should make.

**Comment by jhpratt** [comment]
Reverted all changes mentioned, aside from the one that I believe is correct.

**Review by lebensterben — APPROVED** [review]
I double checked each changes and they look good to me.

File: text/2795-format-args-implicit-identifiers.md [hunk 6]
```diff
@@ -332,7 +332,7 @@ This syntax is widely used and clear to read. It's [introduced in the Rust Book
 
 ## Other languages
 
-A number of languages support string-interpolation functionality with similar syntax to what Rust's formatting macros. The RFC author's influence comes primarily from Python 3's "f-strings" and Javscript's backticks.
+A number of languages support string-interpolation functionality with similar syntax to what Rust's formatting macros. The RFC author's influence comes primarily from Python 3's "f-strings" and Javascript's backticks.
```
Comment (apiraino) [inline text/2795-format-args-implicit-identifiers.md:335]:
```suggestion
A number of languages support string-interpolation functionality with similar syntax to what Rust's formatting macros. The RFC author's influence comes primarily from Python 3's "f-strings" and JavaScript's backticks.
```

File: text/2645-transparent-unions.md [hunk 7]
```diff
@@ -219,7 +219,7 @@ In summary, many of the questions regarding `#[repr(transparent)]` on a `union`
...
+The role of `#[repr(transparent)]` in nonnull-style optimizations is not entirely clear. Specifically, it is unclear whether the user can rely on these optimizations to be performed when they make a type transparent. [Transparent `union`s somewhat complicate the matter](https://github.com/rust-lang/rfcs/pull/2645#issuecomment-470699497). General consensus seems to be that the compiler is free to decide where and when to perform nonnull-style optimizations on `union`s (regardless of whether or not the `union` is transaprent), and no guarantees are made to the user about when and if those optimizations will be applied. It is still an open question exactly what guarantees (if any) Rust makes about transparent `struct`s (and `enum`s) and nonnull-style optimizations.
```
Comment (apiraino) [inline text/2645-transparent-unions.md:222]:
```suggestion
The role of `#[repr(transparent)]` in nonnull-style optimizations is not entirely clear. Specifically, it is unclear whether the user can rely on these optimizations to be performed when they make a type transparent. [Transparent `union`s somewhat complicate the matter](https://github.com/rust-lang/rfcs/pull/2645#issuecomment-470699497). General consensus seems to be that the compiler is free to decide where and when to perform nonnull-style optimizations on `union`s (regardless of whether or not the `union` is transparent), and no guarantees are made to the user about when and if those optimizations will be applied. It is still an open question exactly what guarantees (if any) Rust makes about transparent `struct`s (and `enum`s) and nonnull-style optimizations.
```

File: text/2195-really-tagged-unions.md [hunk 8]
```diff
@@ -422,7 +422,7 @@ There are a few enum repr combinations that are left unspecified under this prop
...
 
-It was [informally decided earilier this year](https://github.com/rust-lang/rust/issues/40029) that `repr(Int)`should have the behaviour this RFC proposes, as it was being partially relied on (in that it supressed dangerous optimizations) and it made sense to the developers. There is even a test in the rust-lang repo that was added to ensure that this behaviour doesn't regress. So this part of the proposal is already implemented and somewhat tested on stable Rust. This RFC just seeks to codify that this won't break in the future.
+It was [informally decided earilier this year](https://github.com/rust-lang/rust/issues/40029) that `repr(Int)`should have the behaviour this RFC proposes, as it was being partially relied on (in that it suppressed dangerous optimizations) and it made sense to the developers. There is even a test in the rust-lang repo that was added to ensure that this behaviour doesn't regress. So this part of the proposal is already implemented and somewhat tested on stable Rust. This RFC just seeks to codify that this won't break in the future.
```
Comment (apiraino) [inline text/2195-really-tagged-unions.md:425]:
```suggestion
It was [informally decided earlier this year](https://github.com/rust-lang/rust/issues/40029) that `repr(Int)`should have the behaviour this RFC proposes, as it was being partially relied on (in that it suppressed dangerous optimizations) and it made sense to the developers. There is even a test in the rust-lang repo that was added to ensure that this behaviour doesn't regress. So this part of the proposal is already implemented and somewhat tested on stable Rust. This RFC just seeks to codify that this won't break in the future.
```

File: text/1535-stable-overflow-checks.md [hunk 9]
```diff
@@ -37,7 +37,7 @@ It's another rather ad-hoc flag for modifying code generation.
 
 Like other such flags, this applies to the entire code unit,
 regardless of monomorphizations. This means that code generation for a
-single function can be diferent based on which code unit its
+single function can be different based on which code unit its
```
Comment (apiraino) [inline text/1535-stable-overflow-checks.md:40]:
```suggestion
single function can be different based on which code unit it's
```

File: text/1317-ide.md [hunk 10]
```diff
@@ -298,5 +298,5 @@ What kind of IPC protocol to use? HTTP is popular and simple to deal with. It's
 platform-independent and used in many similar pieces of software. On the other
 hand it is heavyweight and requires pulling in large libraries, and requires
 some attention to security issues. Alternatives are some kind of custom
-prototcol, or using a solution like Thrift. My prefernce is for HTTP, since it
+prototcol, or using a solution like Thrift. My preference is for HTTP, since it
```
Comment (apiraino) [inline text/1317-ide.md:301]:
```suggestion
protocol, or using a solution like Thrift. My preference is for HTTP, since it
```

**Review by apiraino — CHANGES_REQUESTED** [review]

**Pull request merged**